qiita_pet/handlers/study_handlers/listing_handlers.py

```diff
@@ -168,10 +174,16 @@ class SearchStudiesAJAX(BaseHandler):
     def get(self, ignore):
         user = self.get_argument('user')
         query = self.get_argument('query')
+        search_type = self.get_argument('search_type')
         echo = int(self.get_argument('sEcho'))
 
         if user != self.current_user.id:
             raise HTTPError(403, 'Unauthorized search!')
+        if search_type not in ['user', 'public']:
+            self.clear()
```

Comment:
I was thinking in the restapi and that this should be the best method to access the list of studies.